compiler/env/OMRKnownObjectTable.cpp

```diff
@@ -115,7 +115,8 @@ OMR::KnownObjectTable::getIndexAt(uintptrj_t *objectReferenceLocation)
 #ifdef J9_PROJECT_SPECIFIC
    TR::VMAccessCriticalSection getIndexCriticalSection(self()->comp());
 #endif
-   uintptrj_t objectPointer = *objectReferenceLocation; // Note: object references held as uintptrj_t must never be compressed refs
+   // Note: object references held as uintptrj_t must never be compressed refs
+   uintptrj_t objectPointer = TR::Compiler->om.getReferenceAtAddress(self()->comp(), (uintptrj_t)objectReferenceLocation);
```

Comment:
Could we use a `static_cast` / `reinterpret_cast` here (similarly elsewhere)?

Reply:
Hm not sure about `reinterpret_cast`, but `static_cast` doesn't like casting from pointer to non-pointer or vice versa.

Reply:
https://ideone.com/Yj8T10